core/src/main/scala/kafka/tools/ConsumerPerformance.scala

```diff
@@ -184,12 +209,21 @@ object ConsumerPerformance {
                            lastMessagesRead: Long,
                            startMs: Long,
                            endMs: Long,
-                           dateFormat: SimpleDateFormat) = {
+                           dateFormat: SimpleDateFormat,
+                           beginTime: Option[Long] = None,
```

Comment:
It's a little weird that we have both a `startTime` and a `beginTime`. I think the reason is that `beginTime` is computed from the start of the test, but `startTime` is computed after the first rebalance. However, now that we are calculating the rebalance time separately, I wonder if we still need to maintain this distinction. Maybe we can just use the test start time for both cases.

Reply:
added `periodicJoinTimeInMs` to track join time for a single interval.